addPeers delPeers registerUser

Balou9 opened the pull request:
Does db.get at deletePeers make sense?

Balou9 committed:
addPeers-delPeers-registerUser test

Balou9 committed:
db add del peers test

chiefbiiko commented on test.js:
```diff
@@ -1007,6 +1007,207 @@ tape('getPeers - fail pt3', t => {
+  const tx = Math.random()
+  const meta_stream = jsonStream(new PassThrough())
+  const metadata = { type: 'REGISTER', user: 'balou',
+                     peers: [], password: 'kd', tx }
```
mach mal object literals
  wenn innerhalb 80 character line limit: **inline**, fx:
  ``` js
  { a: 1, b: 2 }
  ```
  wenn über 80 character line limit: **one prop per line**, fx:
  ``` js
  {
    a: 123456780099458763483637497347654686584658743646874756438568743658746587346,
    b: 435495745658638774544534533423442553665658458783475475365743563845734857542
  }
  ```

chiefbiiko replied:
mir ist gerade aufgefallen, dass `valid.schema_REGISTER` nicht sicherstellt, dass `metadata.status` provided ist. ich push gleich kleinen change...
nimm mal bitte die änderungen bei dir auf..
**dann läuft dein Test nicht mehr durch**, da müsstest du dann bitte noch irgendein `status` string ins `metadata`  objekt setzen müssen, dann läuft der test wieder.

chiefbiiko commented on test.js:
```diff
@@ -1007,6 +1007,207 @@ tape('getPeers - fail pt3', t => {
   })
 })
 
+tape('register user - pass', t => {
```
register user -> registerUser

chiefbiiko commented on test.js:
```diff
@@ -1007,6 +1007,207 @@ tape('getPeers - fail pt3', t => {
+  meta_stream.once('data', res => {
+    t.true(valid.schema_RESPONSE(res), 'response is valid schema R')
+    t.true(res.ok, 'response status ok')
+    t.equal(res.tx, tx, 'transaction identifiers equal')
```
mach mal hier bitte noch eine ergänzung, und zwar ziemlich so:
``` js
db.get(metadata.user, (err, user) => {
  if (err) t.end(err)
  t.equal(user.password, metadata.password, 'password')
  t.equal(user.status, metadata.status, 'status')
  t.same(user.peers, metadata.peers, 'peers')
})
```

chiefbiiko commented on test.js:
```diff
@@ -1007,6 +1007,207 @@ tape('getPeers - fail pt3', t => {
+  const registerUser = createRegisterUser(db)
+  const tx = Math.random()
+  const meta_stream = jsonStream(new PassThrough())
+  const metadata = { type: 'REGISTER', user: 'balou', peers: [], tx }
```
with the new `valid.schema_REGISTER` set `metadata.status` so it really  fails like the comments state

chiefbiiko commented on test.js:
```diff
@@ -1007,6 +1007,207 @@ tape('getPeers - fail pt3', t => {
+  const tx = Math.random()
+  const meta_stream = jsonStream(new PassThrough())
+  const metadata = { type: 'REGISTER', user: 'balou',
+                     peers: [], password: 'kd', tx }
```
set some `metadata.status`

chiefbiiko commented on test.js:
```diff
@@ -1007,6 +1007,207 @@ tape('getPeers - fail pt3', t => {
+  const addPeers = createAddPeers(db)
+  const tx = Math.random()
+  const meta_stream = jsonStream(new PassThrough())
+  const metadata = { type: 'ADD_PEERS', user: 'noop', peers: [], tx }
```
fill `metadata.peers` to make the test meaningful!

chiefbiiko commented on test.js:
```diff
@@ -1007,6 +1007,207 @@ tape('getPeers - fail pt3', t => {
+    db.put('noop', { peers: [], status: 'busy' }, err => {
+      if (err) t.end(err)
+      db.put('og', { peers: [], status: 'busy' }, err => {
+        if (err) t.end(err)
```
the test does not require to have three users in `db` so get rid of all but one `db.put`.
make sure the user object u `put` has props `password`, `peers` and `status` (same as above)

chiefbiiko commented on test.js:
```diff
@@ -1007,6 +1007,207 @@ tape('getPeers - fail pt3', t => {
+  const deletePeers = createDeletePeers(db)
+  const tx = Math.random()
+  const meta_stream = jsonStream(new PassThrough())
+  const metadata = { type: 'DEL_PEERS', user: 'noop', peers: [], tx }
```
fill `metadata.peers`!

chiefbiiko commented on test.js:
```diff
@@ -1007,6 +1007,207 @@ tape('getPeers - fail pt3', t => {
+    db.put('noop', { peers: [], status: 'busy' }, err => {
+      if (err) t.end(err)
+      db.put('og', { peers: [], status: 'busy' }, err => {
+        if (err) t.end(err)
```
only keep the `db.put` for `metadata.user`

chiefbiiko commented on test.js:
```diff
@@ -1007,6 +1007,207 @@ tape('getPeers - fail pt3', t => {
+     db.put('og', { peers: [], status: 'busy' }, err => {
+       if (err) t.end(err)
+       db.put('mikey', { peers: [ 'noop',  'balou' ], status: 'busy' }, err => {
+         if (err) t.end(err)
```
trash some `db.put` s

Balou9 committed:
format

Balou9 committed:
Merge branch 'master' of https://github.com/chiefbiiko/user-media-plug into addPeers-delPeers-registerUser

Balou9 committed:
fixes

Balou9 committed:
sort metadata

chiefbiiko committed:
reformat and minimal updates

tosses some unnecessary props from `metadata` objects and minimally changes addPeers, delPeers, registerUser tests
